GDJS/Runtime/pixi-renderers/layer-pixi-renderer.js

```diff
@@ -236,9 +266,90 @@ gdjs.LayerPixiRenderer.prototype.enableEffect = function(name, value) {
+gdjs.LayerPixiRenderer.prototype.isEffectEnabled = function (name) {
   var filter = this._filters[name];
   if (!filter) return false;
 
   return gdjs.PixiFiltersTools.isEffectEnabled(filter);
 };
+
+/**
+ * Updates the render texture, if it exists.
+ * Also, render texture is cleared with a specified clear color.
+ */
+gdjs.LayerPixiRenderer.prototype._updateRenderTexture = function () {
+  if (!this._pixiRenderer) return;
+
+  if (!this._renderTexture) {
+    this._oldWidth = this._pixiRenderer.screen.width;
+    this._oldHeight = this._pixiRenderer.screen.height;
+
+    var width = this._oldWidth;
+    var height = this._oldHeight;
+    var resolution = this._pixiRenderer.resolution;
+    this._renderTexture = PIXI.RenderTexture.create({
+      width,
+      height,
+      resolution,
+    });
+    this._renderTexture.baseTexture.scaleMode = PIXI.SCALE_MODES.LINEAR;
+  }
+
+  if (
+    this._oldWidth !== this._pixiRenderer.screen.width ||
+    this._oldHeight !== this._pixiRenderer.screen.height
+  ) {
+    this._renderTexture.resize(
+      this._pixiRenderer.screen.width,
+      this._pixiRenderer.screen.height
+    );
+    this._oldWidth = this._pixiRenderer.screen.width;
+    this._oldHeight = this._pixiRenderer.screen.height;
+  }
+
+  var oldRenderTexture = this._pixiRenderer.renderTexture.current;
+  var oldSourceFrame = this._pixiRenderer.renderTexture.sourceFrame;
+
+  this._pixiRenderer.renderTexture.bind(this._renderTexture);
+  this._pixiRenderer.renderTexture.clear(this._clearColor);
+
+  this._pixiRenderer.render(this._pixiContainer, this._renderTexture, false);
+  this._pixiRenderer.renderTexture.bind(
+    oldRenderTexture,
+    oldSourceFrame,
+    undefined
+  );
+};
+
+/**
+ * Get the render texture used to display this layer. If render texture is not used, it is created.
+ * @returns {?PIXI.RenderTexture} RenderTexture of the container.
+ */
+gdjs.LayerPixiRenderer.prototype.getRenderTexture = function () {
+  if (!this._renderTexture) this._updateRenderTexture();
+  return this._renderTexture;
+};
+
+/**
+ * Enable the user of a PIXI.RenderTexture to render the PIXI.Container 
+ * of the layer and, in the scene PIXI container, replace the container 
+ * of the layer by a sprite showing this texture.
+ * @private used only in lighting for now as the sprite could have MULTIPLY blend mode.
+ */
+gdjs.LayerPixiRenderer.prototype._replaceContainerWithSprite = function () {
+  if (!this._pixiRenderer) return;
+
+  this._updateRenderTexture();
+  if (!this._renderTexture) return;
+  this._lightingSprite = new PIXI.Sprite(this._renderTexture);
+  this._lightingSprite.blendMode = PIXI.BLEND_MODES.MULTIPLY;
+
+  var sceneContainer = this._runtimeSceneRenderer.getPIXIContainer();
+  var index = sceneContainer.getChildIndex(this._pixiContainer);
+  sceneContainer.addChildAt(this._lightingSprite, index);
+  sceneContainer.removeChild(this._pixiContainer);
+};
+
+gdjs.LayerPixiRenderer.prototype.getPIXIContainer = function () {
```

Comment:
Are you using this? There is already `getRendererObject`. If possible, it would actually be better to rename this `getRendererObject` to `getPIXIContainer`, because `getRendererObject` is used knowing that a container is returned.

Reply:
Yup, this isn't used anywhere at all as far as I know, it should be dead code then 😬. For now, let's remove this, we could always add it later if required :)

Reply:
Oh right, `getRendererObject` already returns the container, we shouldn't be using this then I guess.